pkg/asset/machines/master.go

```diff
@@ -54,59 +58,56 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 
 	ic := installconfig.Config
 	pool := masterPool(ic.Machines)
-	numOfMasters := int64(0)
-	if pool.Replicas != nil {
-		numOfMasters = *pool.Replicas
-	}
-
 	switch ic.Platform.Name() {
 	case "aws":
-		config := aws.MasterConfig{}
-		config.ClusterName = ic.ObjectMeta.Name
-		config.Region = ic.Platform.AWS.Region
-		config.Machine = defaultAWSMachinePoolPlatform()
-
-		tags := map[string]string{
-			"tectonicClusterID": ic.ClusterID,
+		mpool := defaultAWSMachinePoolPlatform()
+		mpool.Set(ic.Platform.AWS.DefaultMachinePlatform)
+		mpool.Set(pool.Platform.AWS)
+		if mpool.AMIID == "" {
```

Comment:
Per above what should we do about this block here. Hive shouldn't duplicate what's going on here. Could you apply this to the machine sets after they're generated?

Reply:
Resolved.